### ISSUE TITLE
Create option to draw branches always straight, instead of stepwise

### ISSUE THREAD

**Issue by yesgrey** [gitextensions/gitextensions#4383]
**Do you want to request a *feature* or report a *bug*?**
Feature

**What is the current behavior?**
Branches are always drawn close to the one at its left, making it stepwise.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
N/A

**What is the expected behavior?**
It would be good to have an option to allow to draw all branches simply straight. This would require more space for drawing all branches, but it would make it a lot simpler to follow each branch, because right now when we have several branches the drawing is stepwise, with lots of corners, which make it quite hard to follow a branch through the history.
As an example, if we have 10 branches, then the drawing would take the space of 10 columns, even if some of the branches do not reach the top of the drawing.  

**Environment you encounter the issue:**

- GitExtensions version: All
- GIT version: -
- OS version: -

**Did this work in previous version of GitExtensions (which)?**
No

**Comment by RussKie** [comment]
So what should I be expecting in my case?
![image](https://user-images.githubusercontent.com/4403806/35219554-343283b0-ffc7-11e7-88ca-99782e7b1178.png)

**Comment by yesgrey** [comment]
I didn't think about the stash case, which I never use, anyway, it should look something like this:
![image](https://user-images.githubusercontent.com/23011691/35230220-c23903d2-ff95-11e7-9ffc-1a3de8b480d5.png)

or like this, if we consider my other suggestion of creating a way of selecting the order of drawing the branches (bitbucket-tweaks set as the 1st to draw):
![image](https://user-images.githubusercontent.com/23011691/35230878-8cfa0106-ff97-11e7-842d-11c503ada441.png)

Basically, the idea is to always have a separate column for each vertical line that cross commits from different branches/history lines.
There is a GUI called Gitkraken that does it the way I suggest, but it's not as good as Git Extensions in all the rest. ;)

**Comment by RussKie** [comment]
Apologies for not being clear. As you can see on my screenshot I have 2 or 3 dozen of branches. Rendering them in a straight line would leave me with very little usable space, wouldn't it?

**Comment by vbjay** [comment]
I'm with @RussKie on this.  We have 
![image](https://user-images.githubusercontent.com/3629489/35245927-e6d14018-ff92-11e7-9f66-40c894028ab1.png)
that can help with showing the trail a specific branch takes.

**Comment by yesgrey** [comment]
Ok, I missed that. ;)

That's why I requested this to be optional... I can understand that people with plenty of branches would prefer the current viewing mode. However, many times some branches refer only to older commits in a certain branch, and in this case they would only take one column all. Furthermore, now with 4k screens available (I have one) even 40 or 50 branches still leave lots of available space... :) but not everyone has that many branches... ;)

The Ctrl+Shift+B is a nice feature but only allows highlighting one branch at a time, and it's not exactly the same thing...

I feel this would be a very nice option to add, if it's not that much work to do...

**Comment by RussKie** [comment]
The biggest challenge is not making it optional, but implementing a different drawing algorithm.

**Comment by yesgrey** [comment]
Yes, I understand that, I only wanted to clarify that I didn't want to replace the current drawing algorithm, which I'm sure many people find it to be their favourite option.

**Comment by yesgrey** [comment]
Could someone please point me out the file(s) where the current drawing algorithm is implemented? This would make my task easier, otherwise I would have to take some time looking for it... Thanks!

**Comment by RussKie** [comment]
> What's the meaning of the tag "pull request only"?

This means that the core team (which is very small) won't be looking at this request any time soon. And if anyone wants this functionality they will need to send a pull-request in.

> Could someone please point me...

I am not certain for I haven't looked at that myself, but think this may be handled by `DvcsGraph`

**Comment by RussKie** [comment]
@spdr870 is this something we want to implement?

**Comment by GintasS** [comment]
Abandoning this because of age and lack of discussion.
